### PR TITLE
AArch64 macOS: Add a call to pthread_jit_write_protect_np()

### PR DESCRIPTION
**Pull request by knn-k**
This commit adds a call to pthread_jit_write_protect_np(0) in
ARM64ImmSymInstruction::generateBinaryEncoding() for AArch64 macOS.

It is needed because methodTrampolineLookup() calls createTrampoline()
when necessary, and that makes the compiler thread lose the permission
for writing to the code buffer.

Signed-off-by: KONNO Kazuhiro <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
Jenkins build aarch64

**Comment by knn-k** [comment]
There is no other place that call `methodTrampolineLookup()` in AArch64 codegen.

File: compiler/aarch64/codegen/ARM64BinaryEncoding.cpp [hunk 1]
```diff
@@ -180,6 +184,12 @@ uint8_t *TR::ARM64ImmSymInstruction::generateBinaryEncoding()
                destination = (intptr_t)cg()->fe()->methodTrampolineLookup(cg()->comp(), symRef, (void *)cursor);
                TR_ASSERT_FATAL(cg()->comp()->target().cpu.isTargetWithinUnconditionalBranchImmediateRange(destination, (intptr_t)cursor),
                                "Call target address is out of range");
+
+#if defined(OSX)
+               // Regain permission for writing to the code buffer
+               // after calling methodTrampolineLookup()
```
Comment (0xdaryl) [inline compiler/aarch64/codegen/ARM64BinaryEncoding.cpp:190]: Could you amend this comment to include why this is the case?  i.e., that `methodTrampolineLookup()` may call `createTrampoline()` which will acquire/release write protection leaving the write permission disabled in this path.

Comment (knn-k) [reply compiler/aarch64/codegen/ARM64BinaryEncoding.cpp]: Updated the comment.

**Comment by knn-k** [comment]
Jenkins build aarch64

**Review by 0xdaryl — APPROVED** [review]
Only a comment was changed.  No need to wait for CI.   Merging.

**Pull request merged**